src/ui/components/battle-info.ts

```diff
@@ -588,11 +588,11 @@ export class BattleInfo extends Phaser.GameObjects.Container {
         this.genderText.setPositionRelative(this.nameText, this.nameText.displayWidth, 0);
       }
 
-      const teraType = pokemon.getTeraType();
+      const teraType = pokemon.teraType;
```

Comment:
```suggestion
      const teraType = pokemon.isTerastallized ? pokemon.teraType : ElementalType.UNKNOWN;
```
(Fixing a bug where Terastallizing fails to update the Tera icon, since Terastallizing no longer changes the Pokemon's `getTeraType()`.)